keps/sig-node/2033-rootless/README.md

```diff
@@ -0,0 +1,818 @@
+challenging to test, should be called out.
+
+All code is expected to have adequate tests (eventually with coverage
+expectations). Please adhere to the [Kubernetes testing guidelines][testing-guidelines]
+when drafting this test plan.
+
+[testing-guidelines]: https://git.k8s.io/community/contributors/devel/sig-testing/testing.md
+-->
+
+Testing needs cgroup v2 CI infra.
+
+The most easiest way to test Rootless Kubernetes would be to run `kind` with Rootless Docker provider.
+
+Alternatively we could use [Usernetes smoke tests](https://github.com/rootless-containers/usernetes/blob/v20201118.0/.github/workflows/main.yaml#L49-L50) as well.
+
+### Graduation Criteria
+
+<!--
+**Note:** *Not required until targeted at a release.*
+
+Define graduation milestones.
+
+These may be defined in terms of API maturity, or as something else. The KEP
+should keep this high-level with a focus on what signals will be looked at to
+determine graduation.
+
+Consider the following in developing the graduation criteria for this enhancement:
+- [Maturity levels (`alpha`, `beta`, `stable`)][maturity-levels]
+- [Deprecation policy][deprecation-policy]
+
+Clearly define what graduation means by either linking to the [API doc
+definition](https://kubernetes.io/docs/concepts/overview/kubernetes-api/#api-versioning)
+or by redefining what graduation means.
+
+In general we try to use the same stages (alpha, beta, GA), regardless of how the
+functionality is accessed.
+
+[maturity-levels]: https://git.k8s.io/community/contributors/devel/sig-architecture/api_changes.md#alpha-beta-and-stable-versions
+[deprecation-policy]: https://kubernetes.io/docs/reference/using-api/deprecation-policy/
+
+Below are some examples to consider, in addition to the aforementioned [maturity levels][maturity-levels].
+
+#### Alpha -> Beta Graduation
+
+- Gather feedback from developers and surveys
+- Complete features A, B, C
+- Tests are in Testgrid and linked in KEP
+
+#### Beta -> GA Graduation
+
+- N examples of real-world usage
+- N installs
+- More rigorous forms of testing—e.g., downgrade tests and scalability tests
+- Allowing time for feedback
+
+**Note:** Generally we also wait at least two releases between beta and
+GA/stable, because there's no opportunity for user feedback, or even bug reports,
+in back-to-back releases.
+
+#### Removing a Deprecated Flag
+
+- Announce deprecation and support policy of the existing flag
+- Two versions passed since introducing the functionality that deprecates the flag (to address version skew)
+- Address feedback on usage/changed behavior, provided on GitHub issues
+- Deprecate the flag
+
+**For non-optional features moving to GA, the graduation criteria must include 
+[conformance tests].**
+
+[conformance tests]: https://git.k8s.io/community/contributors/devel/sig-architecture/conformance-tests.md
+-->
+
+
+- Alpha: Basic support for rootless mode on cgroups v2 hosts.
+
+- Beta: e2e tests coverage.
```

Comment:
to move to beta, we need clarity if we intend to define two separate types of conformance suites:

- kubernetes clusters that can run privileged workloads
- kubernetes cluster that are restricted to run unprivileged workloads only

absent that distinction i do not see how we proceed to beta or navigate the web of questions and caveats around what works, what doesnt, and what requires testing across the project and related sigs (node, storage, network, etc.)

Reply:
Updated
```md
- Beta: e2e tests coverage.
  Requires [the cgroup v2 KEP](../20191118-cgroups-v2.md ) to reach Beta or GA.
  To move to beta, we need clarity if we intend to define two separate types of conformance suites:
  - kubernetes clusters that can run privileged workloads
  - kubernetes cluster that are restricted to run unprivileged workloads only
 ```